src/main/java/randoop/generation/Bloodhound.java

```diff
@@ -0,0 +1,187 @@
+package randoop.generation;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import randoop.main.CoverageTracker;
+import randoop.operation.TypedOperation;
+import randoop.util.ArrayListSimpleList;
+import randoop.util.Randomness;
+
+/**
+ * Implements the Bloodhound component, largely as described by the authors of the Guided Random
+ * Testing paper. Bloodhound computes a weight for each method under test by taking a weighted
+ * combination of the uncovered branch ratio and the ratio between the number of times the method
+ * has been invoked and the number of times any method under test has been invoked.
+ */
+public class Bloodhound {
+  /**
+   * Map of methods under test to their weights. These weights are dynamic and depend on branch
+   * coverage.
+   */
+  private final Map<TypedOperation, Double> methodWeights = new HashMap<>();
+
+  /**
+   * Map of methods under test to the number of times they have been selected for the new sequence.
+   * Cleared every time coverage is recomputed.
+   */
+  private final Map<TypedOperation, Integer> methodSelections = new HashMap<>();
+
+  /** Map of methods under test to the number of times they have been successfully invoked. */
+  private final Map<TypedOperation, Integer> methodSuccCalls = new HashMap<>();
+
+  /**
+   * Simple list of operations, identical to ForwardGenerator's operation list. Needed for getting
+   * weighted member.
+   */
+  private final ArrayListSimpleList<TypedOperation> operationSimpleList =
+      new ArrayListSimpleList<>();
+
+  /** Hyper-parameter for balancing branch coverage and number of time a method was chosen. */
+  private final double alpha = 0.7;
+
+  /** Hyper-parameter for decreasing weights. */
+  private final double p = 0.5;
+
+  /** Maximum number of successful calls to any method under test. */
+  private int maxSuccessfulCalls = 0;
+
+  /** Step number used to determine when to recompute method weights. */
+  private int stepNum = 0;
+
+  /**
+   * Copy the provided list of operations into bloodhound's simple list.
+   *
+   * @param operations list of operations to copy.
+   */
+  public void setOperations(List<TypedOperation> operations) {
+    // Clear out the simple list of operations.
+    operationSimpleList.theList.clear();
+
+    // Copy every operation into our simple list.
+    for (TypedOperation operation : operations) {
+      operationSimpleList.add(operation);
+    }
+  }
+
+  /**
+   * Recompute weights for all methods under test at regular intervals. Each method under test is
```

Comment:
This documentation can be improved.  This method doesn't do anything at regular intervals.  Is it called at regular intervals?  Does this method always do recomputation, or just sometimes?  Please improve the documentation.